module_utils/ios/config/l2_interfaces/l2_interfaces.py

```diff
@@ -0,0 +1,338 @@
+                if state == 'deleted':
+                    commands.extend(L2_Interfaces._state_deleted(**kwargs))
+
+                if state == 'merged':
+                    commands.extend(L2_Interfaces._state_merged(**kwargs))
+
+                if state == 'replaced':
+                    commands.extend(L2_Interfaces._state_replaced(**kwargs))
+
+        return commands
+
+    @staticmethod
+    def _state_replaced(**kwargs):
+        """ The command generator when state is replaced
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :param interface_type: interface type
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the deisred configuration
+        """
+        commands = []
+        want = kwargs['want']
+        obj_in_have = kwargs['have']
+        name = want['name']
+
+        if name:
+            interface = 'interface ' + name
+        if obj_in_have:
+            for item in L2_Interfaces.params:
+                candidate = want.get(item)
+                if item == 'trunk' and candidate:
+                    for each in L2_Interfaces.trunk_params:
+                        if each == 'allowed_vlan':
+                            if candidate[0].get(each):
+                                proposed = (",".join(map(str, candidate[0].get(each)))).split(',')
+                            value = obj_in_have.get(each).split(',')
+                        else:
+                            proposed = candidate[0].get(each)
+                            value = obj_in_have.get(each)
+                        if value and proposed != value:
+                            L2_Interfaces._remove_command_from_interface(interface, L2_Interfaces.trunk_cmds.get(each),
+                                                                         commands)
+        commands.extend(L2_Interfaces._state_merged(**kwargs))
+
+        return commands
+
+    @staticmethod
+    def _state_overridden(**kwargs):
+        """ The command generator when state is overridden
+        :param want: the desired configuration as a dictionary
+        :param obj_in_have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        want = kwargs['want']
+        obj_in_have = kwargs['have']
+        interface_want = 'interface ' + want[0]['name']
+
+        for have in obj_in_have:
+            name = have['name']
+            obj_in_want = search_obj_in_list(name, want)
+            if not obj_in_want:
+                interface_type = get_interface_type(name)
+                if interface_type.lower() == 'loopback':
+                    commands.append('interface ' + name)
+                    commands.append('no description')
```

Comment:
Why do this here?

Reply:
ah.. my bad checked in the wrong file, fixed now. Thanks!